src/dos/drive_virtual.cpp

```diff
@@ -636,6 +602,23 @@ const char* Virtual_Drive::GetLabel()
 	return "DOSBOX";
 }
 
+bool Virtual_Drive::equal_name(vfile_block_t cur_file, const char* search_name)
+{
+	unsigned int onpos = cur_file->onpos;
+	return strcasecmp(search_name,
+	                  (std::string(onpos ? vfilenames[onpos].shortname +
+	                                               std::string(1, '\\')
+	                                     : "") +
+	                   cur_file->name)
+	                          .c_str()) == 0;
+}
+vfile_block_t Virtual_Drive::find_file_by_name(const char* search_name)
```

Comment:
What does this function use from the `Virtual_Drive::` class? Can it be moved out of the class and into a local static helper function?